### DISCUSSION TITLE
Hi Pit seb

### DISCUSSION THREAD

**Discussion by GabrilXX** [ptitSeb/box64#905] [General]
I am developing a project using your box as a base but pulse audio is not working,Would you be able to help me?

**Comment by ptitSeb** [comment]
> I am developing a project using your box as a base but pulse audio is not working,Would you be able to help me?

Maybe. What is not working? Is there some missing symbol or it just crash or something else?

**Comment by GabrilXX** [comment]
Hum ok,I and some friends of mine are developing a project that aims to run
wine compiled x86 on android, but we have already managed to run wine32 on
box86 natively, so we have an impasse that and the audio it does not load
libpulse.so and how could we port box64 to have a build on android?

Em seg., 24 de jul. de 2023 6:42 PM, ptitSeb ***@***.***>
escreveu:

**Reply by ptitSeb** [reply]
So you want to build box64 for android? like box86 can be build for Android target? Or your current version of box64 doesn't want to load the libpulse.so for wine?

**Comment by GabrilXX** [comment]
error log

Box86 with Dynarec v0.3.1 8335a8c2 built on Jul 22 2023 13:41:25 BOX86:
Wine detected, WINEPRELOADRESERVE="000400000-00051a000" Warning, older than
2.34 pthread_kill not found, using current one
Using default BOX86_LD_LIBRARY_PATH: ./:lib/:lib32/:x86/:i686/ Using
default BOX86_PATH: ./:bin/ Counted 34 Env var
Looking for /data/data/com.termux/files/home/x86/lib/wine/../../bin/wine
argv[1]="winecfg" Warning, PreInit Array (size=8) present and ignored!
Rename process to "wine" Using native(wrapped) libdl.so
Using native(wrapped) libc.so Using emulated
/data/data/com.termux/files/home/x86/bin/../lib/wine/i386-unix/ntdll.so
Using emulated
/data/data/com.termux/files/home/x86/lib/wine/i386-unix/win32u.so Using
native(wrapped) libm.so
Using native(wrapped) libfreetype.so Using emulated
/data/data/com.termux/files/home/x86/lib/wine/i386-unix/winex11.so Using
native(wrapped) libXext.so Using native(wrapped) libX11.so.6 Using
native(wrapped) libxcb.so.1 Using native(wrapped) libXau.so.6 Using
native(wrapped) libXdmcp.so.6
Using emulated
/data/data/com.termux/files/home/x86/lib/wine/i386-unix/winspool.so
Using emulated
/data/data/com.termux/files/home/x86/lib/wine/i386-unix/winepulse.so Error
loading needed lib libpulse.so Error loading one of needed lib
Error initializing needed lib winepulse.so
00d8:err:mmdevapi:DllGetClassObject Driver initialization failed
00d8:err:ole:apartment_getclassobject DllGetClassObject returned error
0x80004005 for dll L"C:\\windows\\system32\\mmdevapi.dll"
00d8:err:ole:com_get_class_object no class object
{bcde0395-e52f-467c-8e3d-c4579291692e} could be created for context 0x1

Em seg., 24 de jul. de 2023 6:54 PM, Jose Gabriel <
***@***.***> escreveu:

**Reply by ptitSeb** [reply]
Mmm, ok. Are you sure libpulse.so for armhf exist on your system? there is no pulseaudio native on Android IIRC, so it must be something from termux.

**Comment by GabrilXX** [comment]
exists but does not load, we are suspicious of the libs but not sure


Em seg., 24 de jul. de 2023 7:01 PM, ptitSeb ***@***.***>
escreveu:

**Reply by ptitSeb** [reply]
maybe do a `file libpusle.so` to check it's the correct lib.

**Reply by ptitSeb** [reply]
also, if you want more logs on loading libs, you can use `BOX86_DLSYM_ERROR=1` for more details on dlopen error (probably what is happening with libpulse).

To get even more details on libs error without having too much logs, you can hack the source, in `library.c` and `librarian.c` look for `printf(LOG_DEBUG, ` and changes all occurence with `printf(/*LOG_DEBUG*/LOG_INFO, ` and rebuild. That should gives lots of details on what is going on.

**Comment by GabrilXX** [comment]
https://t.me/+pPyPuVNbbMw1OTFh

Telegram group in case you have

Em seg., 24 de jul. de 2023 7:01 PM, ptitSeb ***@***.***>
escreveu:

**Reply by ptitSeb** [reply]
ah telegram. Don't have an account for that, sorry.

**Comment by GabrilXX** [comment]
Ok, no problem.

Em seg., 24 de jul. de 2023 7:11 PM, ptitSeb ***@***.***>
escreveu:

**Comment by GabrilXX** [comment]
Do you have another means of contact? other than by email?

Em seg., 24 de jul. de 2023 7:11 PM, ptitSeb ***@***.***>
escreveu:

**Reply by ptitSeb** [reply]
I'm on discord
also on matrix

**Comment by GabrilXX** [comment]
I sent you an invite.

Em seg., 24 de jul. de 2023 7:20 PM, ptitSeb ***@***.***>
escreveu:

**Reply by ptitSeb** [reply]
I received 2 invites?

**Comment by GabrilXX** [comment]
Yes, I did.

Em seg., 24 de jul. de 2023 7:29 PM, ptitSeb ***@***.***>
escreveu:

**Comment by GabrilXX** [comment]
I am mcfly ok

Em seg., 24 de jul. de 2023 7:29 PM, Jose Gabriel <
***@***.***> escreveu:
